pins/tests/test_boards.py

```diff
@@ -103,6 +123,101 @@ def test_board_pin_write_feather_deprecated(board):
         board.pin_write(df, "cool_pin", type="feather")
 
 
+def test_board_pin_write_file_raises_error(board, tmp_path):
+    df = pd.DataFrame({"x": [1, 2, 3]})
+
+    path = tmp_path.joinpath("data.csv")
+    df.to_csv(path, index=False)
+
+    # TODO: should this error?
+    with pytest.raises(NotImplementedError):
+        board.pin_write(path, "cool_pin", type="file")
+
+
+def test_board_pin_download(board_with_cache, tmp_path):
+    # create and save data
+    df = pd.DataFrame({"x": [1, 2, 3]})
+
+    path = tmp_path / "data.csv"
+    df.to_csv(path, index=False)
+
+    meta = board_with_cache.pin_upload(path, "cool_pin")
+    assert meta.type == "file"
+
+    (pin_path,) = board_with_cache.pin_download("cool_pin")
+    df = pd.read_csv(pin_path)
+    assert df.x.tolist() == [1, 2, 3]
+
+    with pytest.raises(NotImplementedError):
+        board_with_cache.pin_read("cool_pin")
```

Comment:
Can we point people to `pin_download` when they try to read a file in this error message? Right now it is just `NotImplementedError: No driver for type file`.